Demo/app/src/main/java/com/krunal/demo/kotlinPractice/collection/CollectionMain.kt

```diff
@@ -0,0 +1,267 @@
+package com.krunal.demo.kotlinPractice.collection
+
+fun main() {
+
+    // Array
+    println("\nArray\n")
+    val array = Array(3) { 0 }
+    array[0] = 1
+    array[1] = 2
+    array[2] = 3
+    println(array.contentToString())
+
+
+    // List
+    println("\nList\n")
+    val numList = mutableListOf<Int>()
+    numList.addAll(array)
+    numList.addAll(listOf(4, 3, 2, 5, 3))
+    println(numList)
+
+    // Iterator
+    println("\nIterator\n")
+    val iterator = numList.iterator().withIndex()
+    while (iterator.hasNext()) {
+        print("${iterator.next().value}: ${iterator.next().value}")
+    }
+    println()
+
+    // Progression
+    println("\nProgression\n")
+    for (i in 1..10 step 2) {
+        print("$i ")
+    }
+    println()
+
+    for (i in 1 until 10 step 2) {
+        print("$i ")
+    }
+    println()
+
+    for (i in 10 downTo 1 step 2) {
+        print("$i ")
+    }
+    println()
+
+    for (i in (10 downTo 1).reversed() step 2) {
+        print("$i ")
+    }
+    println()
+
+    // Sequence
+    println("\nSequence\n")
+    var charSeq = sequenceOf('a', 'b', 'c', 'd')
+    charSeq = ('a'..'k').asSequence()
+    charSeq.filter { it in listOf('a', 'e', 'i', 'o', 'u') }.map { print(it); "$it" }.count()
+    println()
+    val oddNumSeq = generateSequence(2) { it + 2 }
+    println(oddNumSeq.take(10).toList())
+    val topTen = generateSequence(1) { if (it < 10) it + 1 else null }
+    println(topTen.toList())
+
+    val oddSeq = sequence {
+        yield(1)
+        yieldAll(listOf(3, 5, 7))
+    }
+    println(oddSeq.toList())
+
+    // Transformation
+    println("\nTransformation\n")
+    val numStrList = listOf("1", "2", "3", "4", "5", "6", "7", "4", "8", "2")
```

Comment:
do not use data type in the variable name, kotlin is strongly typed language, so data type in the variable name is redundant.

Reply:
also use full name `numberStrings`